### DISCUSSION TITLE
Building rendering glitch

### DISCUSSION THREAD

**Discussion by PhungVanHoa** [mapsforge/vtm#954] [General]
Some buildings render incorrectly.
it happen on both S3DBLayer and BuildingLayer (mesh or not). i try on android (with gdx and pure Android) and iOS, the result is same

**Overpass Source: overpass-api.de/api/interpreter?data=[out:json];**
[Video](https://www.dropbox.com/s/7k2lmepdlcknreg/Overpass.mp4?dl=0)

**OSM Building Source: data.osmbuildings.org/0.2/anonymous/tile**
[Video](https://www.dropbox.com/s/qcx2n5p2x7qgkke/osmbuildings.mp4?dl=0)

Coordinates for testing: lat: 52.2340124, lng: 21.0030066
may be related to [this](https://github.com/mapbox/mapbox-gl-js/issues/7766)

Thank you!

**Comment by devemux86** [comment]
Most of the testing is done with Mapsforge offline maps and `BuildingLayer` on (pure) Android and Desktop (libGDX).

On Android you can try OpenGL 3 (see [MapView](https://github.com/mapsforge/vtm/blob/master/vtm-android/src/org/oscim/android/MapView.java#L71)).

**Reply by PhungVanHoa** [reply]
Do you know how to fix this?
I also face problem with rendering text label on iOS (label has white background), should i make a new issue for this?

**Reply by PhungVanHoa** [reply]
> Most of the testing is done with Mapsforge offline maps and `BuildingLayer` on (pure) Android and Desktop (libGDX).
> 
> On Android you can try OpenGL 3 (see [MapView](https://github.com/mapsforge/vtm/blob/master/vtm-android/src/org/oscim/android/MapView.java#L71)).

OpenGL 3 also gives the same result :(

**Comment by devemux86** [comment]
_For questions, reports and feature requests please use the Discussions._

---

None seem to work on iOS or browser, focus is on Android and Desktop.
The buildings are more for show, so not many users work on them either.

(such things depend also on the graphics drivers on each device, see [GLAdapter](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/backend/GLAdapter.java#L32-L40))

**Reply by PhungVanHoa** [reply]
it's sad that no one works on iOS and browser. This library is awesome!
Next time i will create a pull request to run vector map source on iOS

**Comment by PhungVanHoa** [comment]
Buildings in Berlin (offline map)

https://user-images.githubusercontent.com/16158238/185275957-92e1b742-082b-46f7-a63a-8a6ec84b878a.mp4


https://user-images.githubusercontent.com/16158238/185275975-82a7952c-a580-4a45-b20e-0b0b5ca6ba78.mp4




https://user-images.githubusercontent.com/16158238/185276345-43aa7358-52c8-4190-b200-06aa705e574b.mp4


https://user-images.githubusercontent.com/16158238/185276393-d7272180-67cd-495c-905a-71169f2663a9.mp4
